catalog/transformer/catalog-transformer-streaming-lib/src/main/java/org/codice/ddf/transformer/xml/streaming/lib/XmlInputTransformer.java

```diff
@@ -260,9 +269,37 @@ public MetacardType getMetacardType() {
             }
         }
         attributeDescriptors.addAll(BasicTypes.BASIC_METACARD.getAttributeDescriptors());
-        return new DynamicMetacardType(attributeDescriptors, id);
+
+        DynamicMetacardType dynamicMetacardType = new DynamicMetacardType(attributeDescriptors, id);
+        registerMetacardType(dynamicMetacardType);
+        return dynamicMetacardType;
+    }
+
+    private void registerMetacardType(MetacardType metacardType) {
```

Comment:
I recommend moving the dynamic metacard type generation from the transform method to the factory configuration update and then register the metacard type there.